packages/noya-renderer/src/components/PseudoPathLine.tsx

```diff
@@ -1,25 +1,50 @@
+import Sketch from '@sketch-hq/sketch-file-format-ts';
+import { CurvePoint } from '@sketch-hq/sketch-file-format-ts/dist/cjs/types';
 import { Point } from 'noya-geometry';
 import { useStroke } from 'noya-react-canvaskit';
 import React, { useMemo } from 'react';
 import { useTheme } from 'styled-components';
-import { Polyline, Primitives } from '..';
+import { Path, Primitives, useCanvasKit } from '..';
+import { DecodedCurvePoint, encodeCurvePoint } from '../primitives/path';
 
 interface EditablePathPointProps {
   point: Point;
-  decodedCurvePoint: Primitives.DecodedCurvePoint;
+  curvePoint: CurvePoint;
+  frame: Sketch.Rect;
 }
 
 export default function PseudoPathLine({
   point,
-  decodedCurvePoint,
+  curvePoint,
+  frame,
 }: EditablePathPointProps) {
+  const CanvasKit = useCanvasKit();
   const { primary } = useTheme().colors;
-
   const stroke = useStroke({ color: primary });
 
+  const decodedPointToDraw: DecodedCurvePoint = {
+    _class: 'curvePoint',
+    cornerRadius: 0,
+    curveFrom: point,
+    curveTo: point,
+    hasCurveFrom: false,
+    hasCurveTo: false,
+    curveMode: Sketch.CurveMode.Straight,
+    point,
+  };
+
+  const encodedPointToDraw = encodeCurvePoint(decodedPointToDraw, frame);
+
   const points = useMemo(() => {
-    return [point, decodedCurvePoint.point];
-  }, [decodedCurvePoint.point, point]);
+    return [encodedPointToDraw, curvePoint];
+  }, [encodedPointToDraw, curvePoint]);
```

Comment:
This memoization won't work as we have it, since we create a new point each render. We can move the `decodedPointToDraw` and `encodedPointToDraw` into the `useMemo` that should work. The `path` can probably go into the same `useMemo`

Reply:
refactored to have this component to take a CurvePoint[]